nupic/research/frameworks/continuous_learning/dend_kwinners.py

```diff
@@ -0,0 +1,162 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import abc
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+
+class DKWinnersBase(nn.Module, metaclass=abc.ABCMeta):
```

Comment:
This file overlaps with dc_k_winners.py - it has many of the same classes. What's the difference?

Reply:
this is for the k=1 k-winner implementation. dc_k_winners.py is to control when the duty cycles are updated. i'll make sure to add a docstring and resolve ambiguity between similar class names